test/test_replaygain.py

```diff
@@ -166,51 +269,137 @@ def test_cli_writes_only_r128_tags(self):
             self.assertIsNotNone(mediafile.r128_track_gain)
             self.assertIsNotNone(mediafile.r128_album_gain)
 
-    def test_target_level_has_effect(self):
-        item = self.lib.items()[0]
+    def test_targetlevel_has_effect(self):
+        album = self._add_album(1)
+        item = album.items()[0]
 
         def analyse(target_level):
             self.config['replaygain']['targetlevel'] = target_level
-            self._reset_replaygain(item)
             self.run_command(u'replaygain', '-f')
-            mediafile = MediaFile(item.path)
-            return mediafile.rg_track_gain
+            item.load()
+            return item.rg_track_gain
 
         gain_relative_to_84 = analyse(84)
         gain_relative_to_89 = analyse(89)
 
-        # check that second calculation did work
-        if gain_relative_to_84 is not None:
-            self.assertIsNotNone(gain_relative_to_89)
+        self.assertNotEqual(gain_relative_to_84, gain_relative_to_89)
+
+    def test_r128_targetlevel_has_effect(self):
+        if not self.has_r128_support:
+            self.skipTest("r128 tags for opus not supported on backend {}"
+                          .format(self.backend))
+
+        album = self._add_album(1, ext="opus")
+        item = album.items()[0]
+
+        def analyse(target_level):
+            self.config['replaygain']['r128_targetlevel'] = target_level
+            self.run_command(u'replaygain', '-f')
+            item.load()
+            return item.r128_track_gain
+
+        gain_relative_to_84 = analyse(84)
+        gain_relative_to_89 = analyse(89)
 
         self.assertNotEqual(gain_relative_to_84, gain_relative_to_89)
 
+    def test_per_disc(self):
+        # Use the per_disc option and add a little more concurrency.
+        album = self._add_album(track_count=4, disc_count=3)
+        self.config['replaygain']['per_disc'] = True
+        self.run_command(u'replaygain', '-a')
+
+        # FIXME: Add fixtures with known track/album gain so that we can
+        # actually check per-disc operation here.
```

Comment:
The different backends would probably give slightly different values so we'd have to check within some tolerance?

Reply:
Updated the comment so when/if someone does this, the tolerance won't be forgotten.